<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>
    SPDX-FileCopyrightText: none
    SPDX-License-Identifier: GPL-3.0-or-later
  </author>
 <class>WdgSprayOptions</class>
 <widget class="QWidget" name="WdgSprayOptions">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>464</width>
    <height>553</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
    <horstretch>255</horstretch>
    <verstretch>255</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>425</width>
    <height>290</height>
   </size>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,1">
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string comment="Name of the group of options related to the spay area in the spray engine">Area</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="diameterLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string comment="Label for the diameter of the spray area in the spray engine">Diameter:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KisDoubleSliderSpinBox" name="diameterSpinBox" native="true"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="aspectLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string comment="Label for the aspect ratio of the spray area in the spray engine">Aspect ratio:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="KisDoubleSliderSpinBox" name="aspectSPBox" native="true"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="rotationLbl">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string comment="Label for the angle of the spray area in the spray engine">Angle:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="KisAngleSelector" name="rotationAngleSelector" native="true"/>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="scaleLbl">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string comment="Label for the scale of the spray area in the spray engine">Scale:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="KisDoubleSliderSpinBox" name="scaleSpin" native="true"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="spacingLabel">
        <property name="text">
         <string comment="Label for the spacing of the spray area in the spray engine">Spacing:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="KisDoubleSliderSpinBox" name="spacingSpin" native="true"/>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="labelJitter">
        <property name="text">
         <string comment="Label for the jitter movement of the spray area in the spray engine">Jitter Movement:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="jitterMoveBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KisDoubleSliderSpinBox" name="jitterMovementSpin" native="true">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string comment="Name of the group of options related to the spay particles in the spray engine">Particles</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string comment="Label for the amount of particles in the spray engine">Amount:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="0,0,1">
        <item>
         <widget class="QRadioButton" name="countRadioButton">
          <property name="text">
           <string comment="Check box to indicate that the user should set the exact number of particles in the spray engine">Count</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="densityRadioButton">
          <property name="text">
           <string comment="Check box to indicate that the user should set the number of particles in terms of density relative to the spray area in the spray engine">Density</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="KisDoubleSliderSpinBox" name="particlesSpinBox" native="true"/>
          </item>
          <item>
           <widget class="KisDoubleSliderSpinBox" name="coverageSpin" native="true"/>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QGroupBox" name="groupBoxAngularDistribution">
          <property name="title">
           <string comment="Name of the group of options related to the angular distribution of particles in the spray engine">Angular</string>
          </property>
          <layout class="QVBoxLayout" name="layoutAngularDist" stretch="0,0,1">
           <property name="leftMargin">
            <number>6</number>
           </property>
           <property name="topMargin">
            <number>6</number>
           </property>
           <property name="rightMargin">
            <number>6</number>
           </property>
           <property name="bottomMargin">
            <number>6</number>
           </property>
           <item>
            <widget class="QComboBox" name="angularDistCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string comment="Uniform distribution in spray engine">Uniform</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string comment="Custom curve based distribution in spray engine">Curve</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="curveAngularDistContainer" native="true">
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="KisCurveWidget" name="curveAngularDistWidget" native="true">
                <property name="minimumSize">
                 <size>
                  <width>100</width>
                  <height>64</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="KisSliderSpinBox" name="curveAngularDistSpin" native="true"/>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBoxRadialDistribution">
          <property name="title">
           <string comment="Name of the group of options related to the radial distribution of particles in the spray engine">Radial</string>
          </property>
          <layout class="QVBoxLayout" name="layoutRadialDist" stretch="0,0,0,0,0,1">
           <property name="leftMargin">
            <number>6</number>
           </property>
           <property name="topMargin">
            <number>6</number>
           </property>
           <property name="rightMargin">
            <number>6</number>
           </property>
           <property name="bottomMargin">
            <number>6</number>
           </property>
           <item>
            <widget class="QComboBox" name="radialDistCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string comment="Uniform distribution in spray engine">Uniform</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string comment="Gaussian distribution in spray engine">Gaussian</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string comment="Clustering based distribution in spray engine">Cluster</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string comment="Custom curve based distribution in spray engine">Curve</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="KisDoubleSliderSpinBox" name="stdDeviationRadialDistSpin" native="true"/>
           </item>
           <item>
            <widget class="KisDoubleSliderSpinBox" name="clusterRadialDistSpin" native="true"/>
           </item>
           <item>
            <widget class="QWidget" name="curveRadialDistContainer" native="true">
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="KisCurveWidget" name="curveRadialDistWidget" native="true">
                <property name="minimumSize">
                 <size>
                  <width>100</width>
                  <height>64</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="KisSliderSpinBox" name="curveRadialDistSpin" native="true"/>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="centerBiasedPolarDistanceBox">
             <property name="text">
              <string comment="Check box to indicate that the radial distribution should use the old method where particles are biased towards the center of the spray area in the spray engine">Center-biased spread (legacy)</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_3">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string comment="Label for the distribution of particles in the spray engine">Distribution:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KisDoubleSliderSpinBox</class>
   <extends>QWidget</extends>
   <header location="global">kis_slider_spin_box.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KisAngleSelector</class>
   <extends>QWidget</extends>
   <header location="global">KisAngleSelector.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KisCurveWidget</class>
   <extends>QWidget</extends>
   <header location="global">kis_curve_widget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KisSliderSpinBox</class>
   <extends>QWidget</extends>
   <header location="global">kis_slider_spin_box.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
